Updated ToC for entire ManageIQ + added already existing topics within ToC

qahmed1998 opened the pull request:
Added following additional topics to ToC:
- Creating a Service in for Virtual Machine Provisioning
- Service Provisioning Using an Orchestration Template
- Assigning a Custom Analysis Profile to a Virtual Machine

This PR is editing the current ToC for ManageIQ. No files in terms of their locations were moved in the process. 3 existing topics (as mentioned above) were added to the ToC.

<!--
1. Describe what this Pull Request does and why you think it is needed.
   If this PR includes UI or CLI changes, please include Before/After screenshots
   If this PR includes performance changes, please include Before/After metrics showing improvement.
-->

<!--
2. If this fixes an existing issue, please specify in `Fixes #<id>` format
   (As described in https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue)
-->

<!--
3. Ask @miq-bot to apply a scope label (bug, enhancement, etc) and any additional reviewers or assignees.
   (As described in https://github.com/ManageIQ/miq_bot#requested-tasks)
   e.g. `@miq-bot add-label label_name`
        `@miq-bot add-reviewer @name`
-->

Fryguy commented:
@qasim-ahmed This PR is going to need a bunch of "prior" entries, so that the page detector can understand how the pages were moved around.  Basically, the way it works is for each of these "moved" sections, just add a new key called "prior" which is the url for the old location of the file.  Here's an example from when we moved one of the locations from morphy to najdorf: https://github.com/ManageIQ/manageiq-documentation/blob/najdorf/_data/site_menu.yml#L42

Fryguy commented:
@qasim-ahmed Can you also update the original post here with maybe a screenshot or a manual type-out of the new layout?

Fryguy commented:
@agrare @jrafanie Please also take a look.

qahmed1998 commented:
Hi @agrare @jrafanie Please take a look when you get a chance.

jrafanie commented:
@qahmed1998 how can I see a before and after?

It's hard to visualize what's moving around and being added/removed.  I'll add my comments but they're on the existing terminology so mostly unrelated to this PR.

jrafanie commented on _data/site_menu.yml:
```diff
@@ -179,65 +190,61 @@ administration:
       desc: Advanced automation methods for ManageIQ Management Engine
+    - title: Scripting Actions in ManageIQ
+      path: "/docs/reference/latest/scripting_actions/index.html"
+      desc: "Real-time, bi-directional process integration for ManageIQ Management Engine"
```
What's the difference between ManageIQ Management Engine and ManageIQ?  I see both terminology here.  Do we need to normalize on one term?

qahmed1998 replied:
FYI @Fryguy

jrafanie replied:
Note, I see this term elsewhere so we can potentially normalize the terms in a followup PR.  It's more obvious here since I see both forms of what I think are the same thing.

agrare replied:
I bet this used to be CloudForms Management Engine.  Probably best to just use `"Real-time, bi-directional process integration for ManageIQ"`

qahmed1998 commented:
> @qahmed1998 how can I see a before and after?
> 
> It's hard to visualize what's moving around and being added/removed. I'll add my comments but they're on the existing terminology so mostly unrelated to this PR.

We can potentially merge and see the ToC?

The comments to existing terminology/content re-write will likely need a new issue. I think there is a doc issue for that. FYI @rwellon

agrare commented:
> We can potentially merge and see the ToC?

I think that defeats the purpose of PR review if we have to merge it to see it, agree with @jrafanie it is hard to review without seeing what has changed easier

qahmed1998 commented:
I have posted screenshots above as well. The ss might help!

agrare commented on _data/site_menu.yml:
```diff
@@ -31,47 +31,58 @@ installation:
       path: "/docs/reference/latest/appliances/upgrading.html"
       desc: "How to upgrade appliances"
 
-configuration:
-  title: Configuration
-  children:
-    - title: Deployment Planning Guide
-      path: "/docs/reference/latest/deployment_planning_guide/index.html"
-      desc: "ManageIQ Deployment Planning Guide"
-    - title: General Configuration
-      path: "/docs/reference/latest/general_configuration/index.html"
-      desc: "ManageIQ General Configuration"
-    - title: High Availability Guide
-      path: "/docs/reference/latest/high_availability_guide/index.html"
-      desc: "ManageIQ High Availability Guide"
-    - title: Appliance Hardening Guide
-      path: "/docs/reference/latest/appliance_hardening_guide/index.html"
-      desc: "ManageIQ Appliance Hardening Guide"
-
```
So Configuration moved under Reference https://github.com/ManageIQ/manageiq-documentation/pull/1806/files#diff-46686079121737af3636a6d1c95d842fc981118049ec6ece21ce578576eebd50R201 ?

qahmed1998 replied:
Yes, this aligns with how these topics are shown in downstream.

agrare commented on _data/site_menu.yml:
```diff
@@ -31,47 +31,58 @@ installation:
+      path: "/docs/reference/latest/appliances/backup_and_restore.html"
+      desc: "How to backup and restore appliances"
+
+visualizing:
+  title: Visualizing infrastructure
```
Hm I'm not sure "visualizing" is how I'd describe this section.  It does include viewing inventory but also e.g. reconfiguring_a_vm_disk.md, refreshing_instances_and_images.md, removing_a_datastore.md, creating_a_vm_snapshot.md, etc...

If anything I'd call this "Inventory" or "Managing Inventory", @Fryguy wdyt?

agrare replied:
Oh actually that's what the section used to be called, https://github.com/ManageIQ/manageiq-documentation/pull/1806/files#diff-46686079121737af3636a6d1c95d842fc981118049ec6ece21ce578576eebd50L71

qahmed1998 replied:
Oh I see, I can make the top level heading `Managing inventory`.

jrafanie commented:
Ok, this is the before... I think I can imagine what it will look like:

![image](https://github.com/user-attachments/assets/210b4f65-e629-4224-90b8-53005badda1d)

qahmed1998 commented:
The changes include, adding new top level headings, moving topics from one heading to another and adding topics which existed but were not part of the ToC. The ToC changes that are in this PR currently aligns with the downstream (AIOps) ToC.

agrare commented on _data/site_menu.yml:
```diff
@@ -31,47 +31,58 @@ installation:
+      desc: "How to create a service for virtual machine provisioning"
+    - title: Assigning a Custom Analysis Profile to a Virtual Machine
+      path: "/docs/reference/latest/assigning_custom_analysis_profile_to_vm/index.html"
+      desc: "How to assign custom analysis profile to a virtual machine"
```
I don't know that most of these fall under "Administration", I do think administration for backup&restore or something like creating local users but these are all more related to managing inventory IMO.

jrafanie replied:
Good point.  I don't know how to word it but one is administration of the application and one is using the application.

Perhaps a high level "using" section with categories...
* Using
  * Authentication...
  * Automation
    * All of the provisioning, service, etc. 
    * Scripting actions ...
  * Policy
    * Monitoring, alerts, reporting...
    * Policies and Profiles guide
  * Manage Infra and Inv.
  * Managing Providers...
* Configuration
  * Assigning a Custom Analysis Profile...
  * Deployment planning guide
  * ...
* Administration
  * Appliance backup/restore
  * installion
  * upgrading
...

jrafanie replied:
Note, I would prefer we move things one at a time if we can't have a good visual before/after screenshots.  Ideally, each commit changes one thing and the PR changes aren't too much so it's easy to review and get merged.  Anything that "feels" like it's a separate change should probably be done in followup PRs so the high level reorganization can go in first.

qahmed1998 replied:
I can make changes to each big heading (section) in a single commit. With regards to Installing, Upgrading, It will make sense for it to be its own heading and not be under Administration.

qahmed1998 replied:
@jrafanie @agrare For the following, Should these go under `Configuration` or `Managing inventory`

- Provisioning Virtual Machines and Hosts
- Service Provisioning Using an Orchestration Template
- Creating a Service in for Virtual Machine Provisioning

jrafanie replied:
> * Provisioning Virtual Machines and Hosts
> * Service Provisioning Using an Orchestration Template
> * Creating a Service in for Virtual Machine Provisioning

I think they would go in using -> automation

agrare commented:
>  The ToC changes that are in this PR currently aligns with the downstream (AIOps) ToC.

Okay it seems like some items might be in the wrong place there then also so lets put them in the best place upstream first :tm: and go from there

qahmed1998 commented:
> > The ToC changes that are in this PR currently aligns with the downstream (AIOps) ToC.
> 
> Okay it seems like some items might be in the wrong place there then also so lets put them in the best place upstream first ™️ and go from there

Yeah sounds good to me.

jrafanie commented on _data/site_menu.yml:
```diff
@@ -1,7 +1,7 @@
-installation:
-  title: Installation
+installing:
+  title: Installing
```
As an aside... See line 10 and blow... Kubernetes is the only container based installation with the rest being appliances.  I wonder if there should be categories for installing on container/podified and appliances.

qahmed1998 replied:
Good idea in my opinion.

qahmed1998 replied:
Added commit for this.

agrare commented:
NOTE I don't see `Automation Management Providers` in the existing TOC, can we add under `Managing Providers` similar to `Configuration Providers`

qahmed1998 commented:
> NOTE I don't see `Automation Management Providers` in the existing TOC, can we add under `Managing Providers` similar to `Configuration Providers`

Do you have the URL/path of the file/content?

agrare commented:
https://www.manageiq.org/docs/reference/latest/managing_providers/#automation-management-providers

qahmed1998 commented:
Okay so from our discussions in Slack, this is what we have as updated ToC.

**Installing**

**Upgrading**

**Using**

- General Configuration

- Monitoring, alerts, reporting

- Policies and Profiles Guide

- Managing Infrastructure and Inventory

- Managing Providers and its sub-topics

- Provisioning Virtual Machines and Hosts

- Scripting Actions in ManageIQ

- Creating a Service in for Virtual Machine Provisioning 

- Service Provisioning Using an Orchestration Template 

- Managing controls (seems the most suitable name to me as a heading)

   - Monitoring, alerts, reporting

   - Policies and Profiles Guide 

- Assigning a Custom Analysis Profile to a Virtual Machine 

**Administration**

- Authentication

- Backup and Restore (appliances)

- Deployment planning guide

- High Availability Guide

- Appliance Hardening Guide

**Integration**

- AWS CloudFormation and OpenStack Heat

- ServiceNow

**APIs**

- REST API

**Reference** 

- Capabilities Matrix

- Methods available for automation

**Troubleshooting (same as it is)**

qahmed1998 commented:
Any final adjustments or am I good to make this as the ToC in the PR?

jrafanie commented:
> Okay so from our discussions in Slack, this is what we have as updated ToC.

Looks good to me for reorganizing things!  We can iterate on it going forward.

qahmed1998 commented:
Updated ToC as we discussed. Please review.

qahmed1998 commented:
Updated ToC as we discussed. Please review.

- Updated the ToC
- Added `Automation Management Providers` topic in ToC
- Changed description of a topic
- Added `Chargeback` topic in ToC

jrafanie commented on _data/site_menu.yml:
```diff
@@ -178,66 +161,99 @@ administration:
       path: "/docs/reference/latest/provisioning_virtual_machines_and_hosts/index.html"
       desc: "Provisioning, workload management, and orchestration in ManageIQ"
-
+    - title: Creating a Service in for Virtual Machine Provisioning
```
is `in` a typo?

qahmed1998 replied:
yeah, fixed.

jrafanie commented on _data/site_menu.yml:
```diff
@@ -178,66 +161,99 @@ administration:
+      desc: "How to create a service for virtual machine provisioning"
+    - title: Service Provisioning Using an Orchestration Template
+      path: "/docs/reference/latest/service_provisioning_using_an_orchestration_template/index.html"
+      desc: "How to create orchestration template to deploy services"
```
missing an article: `"How to create an orchestration template to deploy services"

jrafanie commented on _data/site_menu.yml:
```diff
@@ -178,66 +161,99 @@ administration:
+      children:
+        - title: Monitoring, Alerts, and Reporting
+          path: "/docs/reference/latest/monitoring_alerts_and_reporting/index.html"
+          desc: "Information on the Cloud Intelligence area of ManageIQ: reporting, RSS, usage, timelines, and chargeback"
```
We should probably 🗑️ RSS if the UI removed it

qahmed1998 replied:
Should RSS be removed from description?

jrafanie replied:
I don't see RSS mentioned in https://www.manageiq.org/docs/reference/latest/monitoring_alerts_and_reporting/index.html

Note, I don't really see usage there either. The toplevel bullets topics could be summarized as dashboards, widgets, reporting, chargeback, optimization, and alerts.

jrafanie replied:
not sure if we dropped usage or we didn't include it in the index: /monitoring_alerts_and_reporting/_topics/to_access_usage_data.md

qahmed1998 replied:
updated the description.

qahmed1998 commented:
@jrafanie @Fryguy are we good to merge?

jrafanie commented on _data/site_menu.yml:
```diff
@@ -1,77 +1,56 @@
+        children:
+           - title: Amazon Web Services (AWS)
+             path: "/docs/reference/latest/installing_on_aws/index.html"
+             desc: "How to install and configure the ManageIQ Management Engine appliance on Amazon Web Services (AWS)"
```
Were we going to remove the `Management Engine` part throughout or is that for a followup PR?

jrafanie replied:
It should just be ManageIQ for the open source product.

jrafanie commented on _data/site_menu.yml:
```diff
@@ -178,66 +161,99 @@ administration:
     - title: Scripting Actions in ManageIQ
       path: "/docs/reference/latest/scripting_actions/index.html"
-      desc: "Real-time, bi-directional process integration for ManageIQ Management Engine"
+      desc: "Real-time, bi-directional process integration for ManageIQ"
```
Ah, we got rid of Management Engine here.  We should do it throughout this page at the very least.  We can drop it elsewhere afterwards.

qahmed1998 commented:
Hi Joe, I have addressed your comments.

jrafanie reviewed (approved):
Looks good to me.  It's hard to see the layout in this format but I think all of the concerns have been addressed.

Fryguy commented:
This PR broke the build 


```

Fryguy commented:
Should be fixed by https://github.com/ManageIQ/manageiq-documentation/pull/1812

jrafanie commented:
Weird it passed in the PR.  😕

Fryguy commented on _data/site_menu.yml:
```diff
@@ -178,66 +161,99 @@ administration:
-troubleshooting:
-  title: Troubleshooting
+administration:
+  title: Administering {{ site.data.product.title_short }}
```
I didn't see this until after merge, but we don't interpolate the site menu, so this is literally present on the website

See https://www.manageiq.org/docs/reference/

qahmed1998 replied:
I'll fix this

jrafanie commented:
> This PR also "broke" the user reference page - https://github.com/ManageIQ/manageiq.org/blob/master/site/docs/reference/index.md?plain=1
> 
> I wonder if instead of hardcoding the sections we can more dynamically build it up. Even so, the `{{}}` stuff won't work there because it's dynamically built.

@qahmed1998 see above
